### DISCUSSION TITLE
Web播放如何设计动画渲染帧率

### DISCUSSION THREAD

**Discussion by KakaSasa2009** [Tencent/libpag#2599] [Q&A] [answered]
![image](https://github.com/user-attachments/assets/d7bebdf7-7d6f-45b6-9dbf-8d9d8739e615)
描述：web平台将一批lottie转成pag后，发现渲染帧率只有24-25，且通过api setMaxFrameRate无法变更
现象：25帧播放，明显感觉动画不流畅

**Comment by kevingpqi123** [comment, marked as answer]
只有当 setMaxFrameRate 小于 文件帧率的时候，才会有效，这里限制的是最大播放帧率
如果想提升文件帧率，需要重新导出 pag 文件，在 AE 中导出修改或者通过 https://pag.io/docs/SDK-migration.html 转换工具，Lottie json 转 pag 的时候是可以设置帧率的， 如果文件在桌面端预览就会卡顿，这种方式是有效的

如果文件在桌面端预览不卡顿，在 web 平台渲染卡顿，那就是性能问题，需要去优化 pag 素材， pag 渲染的性能和素材复杂度强相关， 上面的方法就没有效果

**Comment by KakaSasa2009** [comment]
![image](https://github.com/user-attachments/assets/d6e59d1f-09be-4921-a447-a900b96b61ac)
现在就是桌面端

**Reply by KakaSasa2009** [reply]
设置了不生效

**Reply by KakaSasa2009** [reply]
![image](https://github.com/user-attachments/assets/4e0dd563-8713-41ea-91db-52dcd3a17d2c)

**Reply by KakaSasa2009** [reply]
这个case是这样的，原始文件是lottie，里面有个字段 fr = 20，即动画播放帧数，但实际上在 web 侧渲染的时候，动画仍然是按照最高频率60渲染的（我理解20帧只用于动画补间计算）；转换到 pag 之后，渲染帧率默认成为 20，造成肉眼可见的卡顿，通过转换工具并不能设置修改
